docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,299 @@
+    - name: suse-chost
+      versions:
+        - version: 15.4
+        - version: 14.4
+        - version: 13.6
+  machineTypes: 
+    - name: m5.large
+      cpu: "4"
+      gpu: "0"
+      memory: 8Gi
+  volumeTypes:
+    - name: gp3
+      class: standard
+      usable: true
+  regions:
+    - name: europe-central-1
+      zones:
+        - name: europe-central-1a
+        - name: europe-central-1b
+        - name: europe-central-1c
+```
+
+and a private cloud profile the private cloud profile from the [manifest section](#manifest).
+
+After the rendering is done, the private cloud profile will look like this:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: PrivateCloudProfile
+metadata:
+  name: private-cloud-profile-xyz
+  namespace: project-xyz
+spec:
+  parent: aws-central-cloud-profile
+  kubernetes:
+    versions:
+      - version: 1.28.6
+        expirationDate: 2024-06-06T01:02:03Z
+  machineImages:
+    - name: suse-chost
+      versions:
+        - version: 16.4
+          expirationDate: 2023-08-8T23:59:59Z
+  machineTypes: 
+    - name: m5.xlarge
+      cpu: "8"
+      gpu: "0"
+      memory: 16Gi
+  volumeTypes:
+    - name: ab6
+      class: premium
+      usable: true
+  regions:
+    - name: europe-special
+      zones:
+        - name: europe-custom-1a
+        - name: europe-custom-1b
+        - name: europe-custom-1c
+status:
+  cloudProfile:
+    apiVersion: core.gardener.cloud/v1beta1
+    kind: CloudProfile
+    metadata:
+      name: private-cloud-profile-xyz
+    spec:
+      type: aws
+      kubernetes:
+        versions:
+          - version: 1.27.1
+          - version: 1.26.3
+          - version: 1.25.8
+          - version: 1.24.6
+          - version: 1.28.6
+            expirationDate: 2024-06-06T01:02:03Z
+      machineImages:
+        - name: suse-chost
+          versions:
+            - version: 16.4
+              expirationDate: 2023-08-8T23:59:59Z
+            - version: 15.4
+            - version: 14.4
+            - version: 13.6
+      machineTypes: 
+        - name: m5.large
+          cpu: "4"
+          gpu: "0"
+          memory: 8Gi
+        - name: m5.xlarge
+          cpu: "8"
+          gpu: "0"
+          memory: 16Gi
+      volumeTypes:
+        - name: ab6
+          class: premium
+          usable: true
+        - name: gp3
+          class: standard
+          usable: true
+      regions:
+        - name: europe-central-1
+          zones:
+            - name: europe-central-1a
+            - name: europe-central-1b
+            - name: europe-central-1c
+        - name: europe-special
+          zones:
+            - name: europe-custom-1a
+            - name: europe-custom-1b
+            - name: europe-custom-1c
+```
+
+The rendering is done by a new custom controller registered to the `gardener-controller-manager`. It a merge conflict is to appear during the rendering process, it should be noted somewhere in the status and the parent cloud profile should be preferred.
```

Comment:
Which negative consequences might this defaulting have (you have a running productive cluster, you change the private cloud profile, make a mistake, and then the default cloud profile becomes active)? Isn't that absolutely disastrous? My gut feeling is, we cannot do that. What happens today btw. if the cloud profile is broken with already existing clusters using it? I would expect that reconciliation fails or whatever, but basically that nothing (bad) happens then (it's stuck until the mistake is rectified).

P.S.: Typo:

```suggestion
The rendering is done by a new custom controller registered to the `gardener-controller-manager`. If a merge conflict is to appear during the rendering process, it should be noted somewhere in the status and the parent cloud profile should be preferred.
```

Reply:
I understand, what is meant by the defaulting is not that the entire cloud profile defaults to the parent cloud profile but rather that when a merge conflict appears, the field/s affected by the merge conflict will be filled using the parent cloud profile. Say a private cloud profile defines a machine type m5.abc that is already defined in the parent cloud profile, for this field, the parent cloud profile should be preferred. However, it could also be defined so that the private cloud profile is preferred.

Reply:
Additionally, merge conflicts would be caught during validation of the private cloud profile and would in most cases fail the validation. However there are exceptions to this rule such as when a private cloud profile defines a machine type that is then later added to the parent cloud profile. In that case, the validation should not fail for the existing private cloud profile.

Reply:
Clarified as of https://github.com/gardener/gardener/pull/9142/commits/0a315107b5884c4f85da8b0339ada1bd030d3cfc.